ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml

```diff
@@ -67,36 +70,47 @@ chain:
-      
-      ICSP_COMMAND=""
-      if [[ $ENABLE_ICSP == "true" ]]; then
-        ICSP_COMMAND=$(echo "--image-content-sources ${SHARED_DIR}/mgmt_icsp.yaml")
-      fi
 
       # We don't have the value of HYPERSHIFT_RELEASE_LATEST when we set CONTROLPLANE_OPERATOR_IMAGE so we
       # have to use a hack like this.
       [[ ${CONTROLPLANE_OPERATOR_IMAGE} = "LATEST" ]] && CONTROLPLANE_OPERATOR_IMAGE="${HYPERSHIFT_RELEASE_LATEST}"
 
       CLUSTER_NAME="$(echo -n $PROW_JOB_ID|sha256sum|cut -c-20)"
       echo "$(date) Creating HyperShift cluster ${CLUSTER_NAME}"
-      /usr/bin/hypershift create cluster aws \
-        ${EXTRA_ARGS} ${ICSP_COMMAND} \
-        --name ${CLUSTER_NAME} \
-        --external-dns-domain=${HYPERSHIFT_EXTERNAL_DNS_DOMAIN} \
-        --node-pool-replicas ${HYPERSHIFT_NODE_COUNT} \
-        --instance-type ${HYPERSHIFT_INSTANCE_TYPE} \
-        --base-domain ${DOMAIN} \
-        --endpoint-access ${ENDPOINT_ACCESS} \
-        --region ${HC_REGION} \
-        --zones ${HC_ZONES} \
-        --control-plane-availability-policy ${HYPERSHIFT_CP_AVAILABILITY_POLICY} \
-        --infra-availability-policy ${HYPERSHIFT_INFRA_AVAILABILITY_POLICY} \
-        --pull-secret=/etc/ci-pull-credentials/.dockerconfigjson \
-        --aws-creds=${AWS_GUEST_INFRA_CREDENTIALS_FILE} \
-        --release-image ${RELEASE_IMAGE} \
-        --control-plane-operator-image=${CONTROLPLANE_OPERATOR_IMAGE:-} \
-        --annotations=hypershift.openshift.io/skip-release-image-validation=true \
-        --additional-tags="expirationDate=$(date -d '4 hours' --iso=minutes --utc)"
+      COMMAND=(
+        /usr/bin/hypershift create cluster aws
+        --name ${CLUSTER_NAME}
+        --external-dns-domain=${HYPERSHIFT_EXTERNAL_DNS_DOMAIN}
+        --node-pool-replicas ${HYPERSHIFT_NODE_COUNT}
+        --instance-type ${HYPERSHIFT_INSTANCE_TYPE}
+        --base-domain ${DOMAIN}
+        --endpoint-access ${ENDPOINT_ACCESS}
+        --region ${HC_REGION}
+        --zones ${HC_ZONES}
+        --control-plane-availability-policy ${HYPERSHIFT_CP_AVAILABILITY_POLICY}
+        --infra-availability-policy ${HYPERSHIFT_INFRA_AVAILABILITY_POLICY}
+        --pull-secret=/etc/ci-pull-credentials/.dockerconfigjson
+        --aws-creds=${AWS_GUEST_INFRA_CREDENTIALS_FILE}
+        --release-image ${RELEASE_IMAGE}
+        --control-plane-operator-image=${CONTROLPLANE_OPERATOR_IMAGE:-}
+        --annotations=hypershift.openshift.io/skip-release-image-validation=true
+        --additional-tags="expirationDate=$(date -d '4 hours' --iso=minutes --utc)" 
+      )
+      
+      if [[ $ENABLE_ICSP == "true" ]]; then
+        COMMAND+=(--image-content-sources "${SHARED_DIR}/mgmt_icsp.yaml")
+      fi
+      
+      if [[ -n $EXTRA_ARGS ]]; then
+        COMMAND+=(${EXTRA_ARGS})
+      fi
+     
+      if [[ $HYPERSHIFT_CREATE_CLUSTER_RENDER == "true" ]]; then
+        "${COMMAND[@]}" --render > "${SHARED_DIR}/hypershift_create_cluster_render.yaml"
```

Comment:
It seems that this is just a workaround. We should create the cluster using the hypershift CLI rather than rendering and modifying the YAML file. If the CLI does not support this, then we should adjust the CLI.

Reply:
@LiangquanLi930 
1) It's not possible to create a day-1 ext-oidc cluster directly with hypershift CLI. 
2) I don't think it's necessary/appropriate to add ext-oidc support to Hypershift CLI. There'd be too many parameters to supply.

Reply:
hmm, Why can't these contents be passed in as a single file?

Reply:
And doing this will make this step feel very strange. Its name is 'create-cluster,' but in reality, it doesn't create a cluster; it just renders it. I'm not sure if this will increase the complexity of this step and make it harder for other users to understand.

Reply:
> hmm, Why can't these contents be passed in as a single file?

Technically, every possible stanza could be supported by the Hypershift CLI, but that would make it overly complex. Honestly, I don't see the necessity for it.

Reply:
> Its name is 'create-cluster,' but in reality, it doesn't create a cluster; it just renders it. 

IMO `create-cluster` corresponds to the name of the subcommand.

Reply:
I still think this will change the semantics of hypershift-aws-create-chain. There is a step to create the cluster, but it doesn't actually create the cluster. Of course, if Seth agrees, then I'm okay with it. 
cc @sjenning

Reply:
@sjenning Need your help to give a clear roadmap here. Do we consider to support OIDC in hypershift cli in future. Or do we have day-2 support on OIDC?

Reply:
@YuLi517 

1) Enabling external OIDC on day-2 is not supported. This is a deliberate decision, see https://issues.redhat.com/browse/OCPBUGS-30028?focusedId=24250543&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-24250543 
2) I don't think it's a good idea to add an option for any sub-stanza (e.g., .spec.configuration.authentication) embedded within the HostedCluster resource. This would make the Hypershift CLI overly complex. However, let's wait for @sjenning's input on this.

Reply:
We do not intend to support OIDC enablement day-2 or enabling a better UX for it in the `hypershift` CLI in the short term.  Rendering out the manifest and modifying is as good a way to do it as any.

Reply:
> I still think this will change the semantics of hypershift-aws-create-chain.
I share this concern but the alternative is to copy-paste the chain into a new chain like `hypershift-aws-render-chain`, that is 90% the same and I don't want to do that either.  Thus, I'm ok with this change.